Sources/TuistKit/Models/Settings.swift

```diff
@@ -55,17 +61,55 @@ class Settings: GraphInitiatable, Equatable {
     /// - Throws: A decoding error if an expected property is missing or has an invalid value.
     required init(dictionary: JSON, projectPath: AbsolutePath, fileHandler: FileHandling) throws {
         base = try dictionary.get("base")
-        let debugJSON: JSON? = try? dictionary.get("debug")
-        debug = try debugJSON.flatMap({ try Configuration(json: $0, projectPath: projectPath, fileHandler: fileHandler) })
-        let releaseJSON: JSON? = try? dictionary.get("release")
-        release = try releaseJSON.flatMap({ try Configuration(json: $0, projectPath: projectPath, fileHandler: fileHandler) })
+        configurations = try dictionary.getArray("configurations").compactMap({ try Configuration(json: $0, projectPath: projectPath, fileHandler: fileHandler) })
     }
 
     // MARK: - Equatable
 
     static func == (lhs: Settings, rhs: Settings) -> Bool {
-        return lhs.debug == rhs.debug &&
-            lhs.release == rhs.release &&
-            lhs.base == rhs.base
+        return lhs.base == rhs.base && lhs.configurations == rhs.configurations
+    }
+}
+
+class TargetSettings: GraphInitiatable, Equatable {
+    
+    public let base: BuildSettings
+    public let buildSettings: [Configuration.Name: BuildSettings]
+    
+    public init(base: BuildSettings, buildSettings: [Configuration.Name: BuildSettings]) {
+        self.base = base
+        self.buildSettings = buildSettings
+    }
+    
+    /// Default constructor of entities that are part of the manifest.
+    ///
+    /// - Parameters:
+    ///   - dictionary: Dictionary with the object representation.
+    ///   - projectPath: Absolute path to the folder that contains the manifest.
+    ///     This is useful to obtain absolute paths from the relative paths provided in the manifest by the user.
+    ///   - fileHandler: File handler for any file operations like checking whether a file exists or not.
+    /// - Throws: A decoding error if an expected property is missing or has an invalid value.
+    required init(dictionary: JSON, projectPath: AbsolutePath, fileHandler: FileHandling) throws {
+        base = try dictionary.get("base")
+        buildSettings = try dictionary.get("buildSettings").mapValues{ try $0.getDictionary() }
+    }
+    
+    // MARK: - Equatable
+    
+    static func == (lhs: TargetSettings, rhs: TargetSettings) -> Bool {
+        return lhs.buildSettings == rhs.buildSettings
+    }
+    
+}
+
+extension JSON {
+    
+    /// Returns a JSON mappable dictionary for self.
+    func getDictionary<T: JSONMappable>() throws -> [String: T] {
```

Comment:
This is a bit of a hack, but we want to remove this mechanism altogether https://github.com/tuist/tuist/issues/236